src/main/java/org/prebid/server/bidder/richaudience/RichaudienceBidder.java

```diff
@@ -0,0 +1,176 @@
+package org.prebid.server.bidder.richaudience;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.iab.openrtb.request.Banner;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Device;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.request.Site;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.richaudience.ExtImpRichaudience;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.util.ObjectUtil;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class RichaudienceBidder implements Bidder<BidRequest> {
+
+    private static final String DEFAULT_CURRENCY = "USD";
+    private static final TypeReference<ExtPrebid<?, ExtImpRichaudience>> RICHAUDIENCE_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public RichaudienceBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final URL url = extractUrl(request);
+        final boolean isSecure = "https".equals(ObjectUtil.getIfNotNull(url, URL::getProtocol));
+        final List<Imp> modifiedImps = new ArrayList<>();
+        boolean isTest = false;
+
+        try {
+            validateRequest(request);
+            for (Imp imp : request.getImp()) {
+                validateImp(imp);
+
+                final ExtImpRichaudience extImpRichaudience = parseImpExt(imp);
+                modifiedImps.add(modifyImp(imp, extImpRichaudience, isSecure));
+
+                isTest |= BooleanUtils.isTrue(extImpRichaudience.getTest());
```

Comment:
Pls reWrite `|=`. We should not use code that can confuse anyone

Reply:
How does compound assignment operator can confuse someone?

Reply:
It can, project code is not a place where you show how "great programmer you are" and how you can do it. It's a place in which other programmers will come and read code. So pls, do it to be easy to read for every one.

Reply:
For your case this one is even not needed just `=` will be enough